de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/spi/impl/DocumentPrintingQueueHandler.java

```diff
@@ -134,8 +138,48 @@ else if (InterfaceWrapperHelper.isInstanceOf(archiveRerencedModel, I_M_InOut.cla
 			final I_M_InOut inout = InterfaceWrapperHelper.create(archiveRerencedModel, I_M_InOut.class);
 			handleInOuts(queueItem, inout);
 		}
+		
+		handleItemName(queueItem, archiveRerencedModel);
 	}
 
+	/**
+	 * Sets the ItemName for each document that we process
+	 * @param queueItem
+	 * @param archiveRerencedModel
+	 */
+	private void handleItemName(final I_C_Printing_Queue queueItem, Object archiveRerencedModel)
+	{
+		// Handles operations specific for invoices.
+		if (InterfaceWrapperHelper.isInstanceOf(archiveRerencedModel, I_C_Invoice.class))
+		{
+			queueItem.setItemName(X_C_Printing_Queue.ITEMNAME_Rechnung);
+		}
+		else if (InterfaceWrapperHelper.isInstanceOf(archiveRerencedModel, I_C_Print_Package.class))
+		{
+			queueItem.setItemName(X_C_Printing_Queue.ITEMNAME_PDF);
+		}
+		else if (InterfaceWrapperHelper.isInstanceOf(archiveRerencedModel, I_M_InOut.class))
+		{
+			queueItem.setItemName(X_C_Printing_Queue.ITEMNAME_VersandWareneingang);
+		}
+		else if (InterfaceWrapperHelper.isInstanceOf(archiveRerencedModel, I_C_Letter.class))
+		{
+			final I_C_Letter letter = InterfaceWrapperHelper.create(archiveRerencedModel, I_C_Letter.class);
+			if (X_C_Printing_Queue.ITEMNAME_Mitgliedsausweis.equals(letter.getAD_BoilerPlate().getName()))
```

Comment:
`AD_BoilerPlate.Name` is a updatable string and could therefor change any time on a user's whim.
Also, this code here strikes me as *very* feature specific...i don't think that leaving i like this is a great idea..

suggestions:
* extract this stuff into a handler that is specific to C_Letter
* introduce a not-updatable `InternalName` column to hold such names